safehttp/incoming_request.go

```diff
@@ -14,9 +14,80 @@
 
 package safehttp
 
-import "net/http"
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+)
 
 // IncomingRequest TODO
 type IncomingRequest struct {
-	req *http.Request
+	req       *http.Request
+	parseOnce sync.Once
+}
+
+// QueryForm parses the query parameters provided in the request. It returns
+// the parsed parameters as a Form object, if no error occured, or the parsing error otherwise.
+func (r *IncomingRequest) QueryForm() (f *Form, err error) {
+	r.parseOnce.Do(func() {
+		if r.req.Method != "GET" {
+			f, err = nil, fmt.Errorf("got request method %s, want GET", r.req.Method)
+		}
```

Comment:
Don't you want to add a return in this if-statement? Like so:
```go
f, err = ...
return
```
And similar in all other ifs in `r.parseOnce.Do` inside of `QueryForm`, `PostForm` and `MultipartForm`

Reply:
Thanks, I missed this bug :)